codegen/module_system.go

```diff
@@ -563,16 +563,27 @@ type ClientsInitGenerator struct {
 func (g *ClientsInitGenerator) Generate(
 	instance *ModuleInstance,
 ) (*BuildResult, error) {
-	clients := readClientDependencySpecs(instance)
+	clients := []*ClientSpec{}
+	instances := sortInstances(instance.ResolvedDependencies["client"])
+
+	for _, instance := range instances {
+		clients = append(clients, instance.GeneratedSpec().(*ClientSpec))
+	}
 
 	includedPkgs := []GoPackageImport{}
-	for _, client := range clients {
+	for i, client := range clients {
 		// TODO: there shouldn't be a special thing for custom
 		if client.ClientType == "custom" {
 			includedPkgs = append(includedPkgs, GoPackageImport{
 				PackageName: client.CustomImportPath,
 				AliasName:   client.ImportPackageAlias,
 			})
+			if len(instances[i].ResolvedDependencies["client"]) > 0 {
+				includedPkgs = append(includedPkgs, GoPackageImport{
+					PackageName: instances[i].PackageInfo.GeneratedPackagePath,
```

Comment:
You shouldn't be using the GeneratedPacakgePath, because some packages are not generated. The ImportPackagePath() method will give you the correct thing, likewise, ImportPackageAlias(). I see why you might be confused about the custom type, really there should be no special treatment of the custom type. We're just going to call a custom initializer that returns a custom type wrapper.